examples/data-objects/inventory-app/src/reactSharedTreeView.tsx

```diff
@@ -0,0 +1,138 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as React from "react";
+import {
+	TreeNodeSchema,
+	TreeConfiguration,
+	ITree,
+	TreeFieldFromImplicitField,
+	TreeView,
+	disposeSymbol,
+	SchemaIncompatible,
+	SharedTree,
+} from "@fluidframework/tree";
+import { DataObject } from "@fluidframework/aqueduct";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { IChannelFactory } from "@fluidframework/datastore-definitions";
+
+/**
+ * This file contains logic not specific to this particular sample that other apps may want to use.
+ * Eventually this should be published as part of a package apps can use.
+ */
+
+/**
+ * TODO: once we add options to factory (for example controlling the write format),
+ * apps will need a way to provide those.
+ */
+export const factory: IChannelFactory = SharedTree.getFactory();
+
+/**
+ * Generic DataObject for shared trees.
+ */
+export abstract class TreeDataObject extends DataObject {
+	#tree?: ITree;
+
+	public get tree(): ITree {
+		if (this.#tree === undefined)
+			throw new Error("view should be initialized by hasInitialized");
+		return this.#tree;
+	}
+
+	protected async initializingFirstTime() {
+		this.#tree = this.runtime.createChannel(undefined, factory.type) as ITree;
```

Comment:
We should be schematizing the tree to initialize it here: I believe current code may race schema updates from multiple clients. We may even want to have runtime or API-level enforcement (somehow, would need to think on design) that a tree's schema is initialized pre-attach: I'd imagine basically all applications want this.

Reply:
With my changes and Taylors fixed schema editing, racing initializes should actually work fine (for a correctly written app) now.

That said, this utility doesn't actually support trying multiple different schema, or making multiple views of the same tree, so at this level of abstraction I guess its fine to bake the config in here, since the error handling from that can (with the new API) be separated. Making this change turned out rather nice, so I have done it.